pkg/insights/insightsclient/insightsclient.go

```diff
@@ -217,6 +217,50 @@ func (c *Client) Send(ctx context.Context, endpoint string, source Source) error
 	return nil
 }
 
+func (c Client) RecvReport(ctx context.Context, endpoint string) (*io.ReadCloser, error) {
+	cv := c.clusterInfo.ClusterVersion()
+	if cv == nil {
+		return nil, ErrWaitingForVersion
+	}
+
+	endpoint = fmt.Sprintf(endpoint, cv.Spec.ClusterID)
+	klog.Infof("Retrieving report for cluster: %s", cv.Spec.ClusterID)
+	klog.Infof("Endpoint: %s", endpoint)
+
+	req, err := http.NewRequest(http.MethodGet, endpoint, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	if req.Header == nil {
+		req.Header = make(http.Header)
+	}
+	req.Header.Set("User-Agent", fmt.Sprintf("insights-operator/%s cluster/%s", version.Get().GitCommit, cv.Spec.ClusterID))
```

Comment:
There is still quite a lot of common code between this and `Send` regarding preparation of the client and headers. 

What if we had a common method `doReq(ctx, req)` that would handle the client and headers preparation,  eliminating the code here to:

```go
// …
req, err := http.NewRequest(http.MethodGet, endpoint, nil)
if err != nil {
    return nil, err
}
resp, err := c.DoReq(ctx, req)
// …
```

Reply:
Can we consider this as "resolved"? I'm just reviewing all the comments that the PR already had to make some clean up